syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -89,8 +100,8 @@ def select_share(
         c_2 = c.share(*players, protocol="falcon", field=2)
         c_L = c.share(*players, protocol="falcon", field=ring_size)
 
-        xor_b_c = FalconHelper.xor(b, c_2).reconstruct()
-        d = c_L * (1 - 2 * xor_b_c) + xor_b_c
+        xor_b_c = FalconHelper.xor(b, c_2.child).reconstruct()
```

Comment:
Q: It doesn't work with c_2? (because now we should have the logic to check if is a wrapper or not)

Reply:
Done